instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/mongo.go

```diff
@@ -39,9 +39,9 @@ func (m *monitor) Started(ctx context.Context, evt *event.CommandStartedEvent) {
 		semconv.DBSystemMongoDB,
 		semconv.DBOperation(evt.CommandName),
 		semconv.DBName(evt.DatabaseName),
-		semconv.NetPeerName(hostname),
-		semconv.NetPeerPort(port),
-		semconv.NetTransportTCP,
+		semconv.NetPeerName(hostname), //nolint:staticcheck
+		semconv.NetPeerPort(port),     //nolint:staticcheck
+		semconv.NetTransportTCP,       //nolint:staticcheck
```

Comment:
These are deprecated semantic conventions. Ignoring a lint check does not seem appropriate here.

Reply:
We can't change these conventions without creating a breaking change. And multiple semantic conventions for a major version [is not supported](https://github.com/open-telemetry/semantic-conventions/issues/1502). The 1.24.0 bump was but it was requested on #6172. See that PR for more information.

Reply:
Those conventions may have been introduced in 1.21.0, but updating would create breaking changes. Resolving that is the purpose of #6172, where 1.26.0 removed the deprecated conventions. Please see that ticket, as the issue it attempts to resolve clearly states the migration constraints: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/6171 .

Reply:
See this branch to see the results: https://github.com/prestonvasquez/opentelemetry-go-contrib/tree/otelmongo-bump-semconv-depr-demo

Reply:
`otelmongo` is not a stable API. We should provide a migration path, but there is not a requirement to maintain backwards compatibility here.

+1 to what @damemi suggests: upgrade to 1.21 and work on a migration strategy from there.

Reply:
@MrAlias otelmongo may not be stable, but what you're asking is still a breaking change. In which case we should just re-open and merge #6172. 

@pellared [specifically asked](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/6172#pullrequestreview-2347227726) that we bump to 1.24.0 in #6172.

Reply:
> Do we want to upgrade to 1.21.0 instead?

Yes.

I imagine @pellared had mentioned 1.24 as that is the compatible version for HTTP and it was assumed the same for messaging.

Reply:
> otelmongo may not be stable, but fixing these deprecations is still a breaking change.

Understood. Hence my comments on thinking though a migration path, but ultimately complying with semver in that this is not a stable API and no stability guarantee has been made.

Reply:
Updated! Thanks for the guidance.